src/Microsoft.Diagnostics.Monitoring/DiagnosticsEventPipeProcessor.cs

```diff
@@ -81,7 +99,17 @@ public async Task Process(int pid, int durationSeconds, CancellationToken token)
                         HandleLoggingEvents(source);
                     }
 
+                    if (_mode == PipeMode.GCDump)
+                    {
+                        // GC
+                        handleEventsTask = HandleGCEvents(source, pid, stopFunc, token);
+                    }
+
                     source.Process();
```

Comment:
if we leave source.Process is there any remaining work? Do we really need to throw cancellation or should we just consider the task done?

Reply:
If the token that is passed into the DiagnosticsMonitor is cancelled, then it will stop the event pipe session. This will cause the EventPipeEventSource.Process method to return, but it does not know why it was stopped, other than having StopProcessing called on the associated session. So we have to either await monitor.CurrentProcessingTask or check the token manually.